index.js

```diff
@@ -120,7 +121,9 @@ class Installer {
         )
         return BB.join(
           this.checkLock(),
-          stat && rimraf(path.join(this.prefix, 'node_modules'))
+          stat && readdirAsync(path.join(this.prefix, 'node_modules')).then(children =>
+            BB.join(children.map(child => rimraf(path.join(this.prefix, 'node_modules', child))))
```

Comment:
You probably want this to be BB.all() instead, since that's the one that awaits an array of promises.

There's also an alternative implementation you can do: since `rimraf` supports `glob` out of the box, `rimraf('foo/*')` should delete the contents of foo without removing the directory itself. It might be slightly easier to read the code that way.